### PR TITLE
Check if milestone and label is added to PR

### PR DESCRIPTION
**Pull request by Czaki**
# Description

This PR improves Label Checker action to run only if `ready to merge` is added and check if labels and milestones are added. 

When adding a Label checker to requires workflows before merge it will enforce adding this information. 

# References

## Type of change
- [x] Maintenance (changes required to run napari, tests, & CI smoothly)

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

File: .github/workflows/label_checker.yml
Comment (Czaki) [inline .github/workflows/label_checker.yml]: @jni - you should like this change as it allows for more than one labels.

**Comment by Czaki** [comment]
As triggers depend on main branch then adding and removing milestone will not trigger this workflow here, but one could play with it by adding/removing milestone/labels.

**Review by Carreau — APPROVED** [review]
Looking forward to try this !

File: .github/workflows/label_checker.yml
Comment (jni) [reply .github/workflows/label_checker.yml]: I do! 😍

File: .github/workflows/label_checker.yml
Comment (jni) [inline .github/workflows/label_checker.yml]: I don't understand why this workflow needs to run when issues change? Or is this because PRs are a type of issue and these categories only apply to issues?

**Review by jni — APPROVED** [review]

File: .github/workflows/label_checker.yml
Comment (Czaki) [reply .github/workflows/label_checker.yml]: For GitHub issues means Issues and Pull Requests (if you check the Rest API, part of operations on PR is done using Issues endpoints). 

There are no milestone-related events only for pull requests.

File: .github/workflows/label_checker.yml
Comment (brisvag) [reply .github/workflows/label_checker.yml]: Not sure, but we might need to follow up on napari/docs to ensure multiple labels are treated properly when generating sections in release notes (i.e: the thing appears multiple times)

**Review by brisvag — APPROVED** [review]
Awesome! One thing I'm not sure about: how does the milestone check even work, and should it be disabled separately?

File: .github/workflows/label_checker.yml
Comment (Czaki) [reply .github/workflows/label_checker.yml]: not in napari/docs, but if double appearance is a problem then it needs to be fixed.

**Comment by Czaki** [comment]
> Awesome! One thing I'm not sure about: how does the milestone check even work, and should it be disabled separately?

miuestone check is GitHub app installed in repository and needs to be uninstalled after merge

File: .github/workflows/label_checker.yml
Comment (brisvag) [reply .github/workflows/label_checker.yml]: I think double appearance is what we want right?

**Comment by Czaki** [comment]
> Would be best if it warns about both things at once.

it will require own code to check this. It is doable but it will be next code to manage. But I you think that it is worth then I may do this. 

> Speaking of which, probably we should rename it to `label and milestone check` just so it's obvious you need to add both as a reviewer!

Rename is last steep as it will prevent from testing it here.

File: .github/workflows/label_checker.yml
Comment (Carreau) [reply .github/workflows/label_checker.yml]: If you remember the early GitHub time, you were able to create and issue and then attached code to it, and it would _become_ a PR. It was a mess becaues then you couldn't close the PR but keep the issue. Since then API have always mixed issues and PRs in a lot of places.

**Review by Carreau — APPROVED** [review]

File: .github/workflows/label_checker.yml
Comment (jni) [reply .github/workflows/label_checker.yml]: I actually don't remember such a time @Carreau, you got in at least one generation ahead of me. 😃  ([update](https://caius.github.io/github_id): about 150K users before me.:smiley:)

**Comment by jni** [comment]
@Czaki From the conversation above, it sounds like you should change the action name before we merge?

**Comment by Czaki** [comment]
Yes. But if I should update name or also file name?

**Comment by jni** [comment]
I think it's nice if the filename matches the action title...

**Comment by Czaki** [comment]
@jni, @brisvag could you check the new name?

**Comment by brisvag** [comment]
Looks good. It's too far down so I can't suggest: can you change also the name of the job itself, and not just the workflow? So if you end up on the `details` page you also see the full name.

**Comment by brisvag** [comment]
Ok, let's get this in!

**Comment by Czaki** [comment]
hm. Adding this to required action before merge add pending state. 

image

It is not faliture, but as I understand @Carreau it is still a problem?

**Comment by jni** [comment]
@Czaki I suggest just removing it from required. Typically maintainers check all red status messages before merging and evaluate whether it's a quick fix etc.

**Comment by Czaki** [comment]
Removed, maybe a number of PR without ready to merge will be small.

**Comment by brisvag** [comment]
If not required, will it be ignored in the context of marking green/red on the PR list?

**Comment by Czaki** [comment]
if not required then chek only impact PR with ready to merge label

**Comment by brisvag** [comment]
> if not required then chek only impact PR with ready to merge label

So right now:
- if there is no `ready to merge` label and everything else passes, the PR is marked as green
- if `ready to merge` is added and does not pass, the PR is marked a red? Or still green?

**Comment by Czaki** [comment]
> if there is no `ready to merge` label and everything else passes, the PR is marked as green

yes

> if `ready to merge` is added and does not pass, the PR is marked a red? Or still green?

red

**Comment by brisvag** [comment]
sounds great then :)

**Pull request merged**
